kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/dhcp/DhcpClientLeases.java

```diff
@@ -0,0 +1,170 @@
+/*******************************************************************************
+ * Copyright (c) 2011, 2017 Eurotech and/or its affiliates
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech
+ *******************************************************************************/
+
+package org.eclipse.kura.linux.net.dhcp;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.kura.KuraErrorCode;
+import org.eclipse.kura.KuraException;
+import org.eclipse.kura.net.IPAddress;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class defines DHCP leases
+ *
+ */
+public class DhcpClientLeases {
+
+    private static final Logger logger = LoggerFactory.getLogger(DhcpClientLeases.class);
+
+    private static final String GLOBAL_DHCP_LEASES_DIR = "/var/lib/dhcp";
+    private static final String IFACE_DHCP_LEASES_DIR = "/var/lib/dhclient";
+
+    private static DhcpClientLeases dhcpClientLeases = null;
+
+    public static synchronized DhcpClientLeases getInstance() {
+        if (dhcpClientLeases == null) {
+            dhcpClientLeases = new DhcpClientLeases();
+        }
+        return dhcpClientLeases;
+    }
+
+    public List<IPAddress> getDhcpGateways(String interfaceName, String address) throws KuraException {
+        return getDhcpGateways(interfaceName, parseHostAddress(address));
+    }
+
+    public List<IPAddress> getDhcpGateways(String interfaceName, IPAddress address) throws KuraException {
+        List<IPAddress> gateways = new ArrayList<>();
+        if (interfaceName == null || interfaceName.isEmpty() || address == null || address.getAddress() == null) {
+            return gateways;
+        }
+
+        List<DhcpClientLeaseBlock> leaseBlocks = getLeaseBlocks(getDhclientFile(interfaceName));
+        for (DhcpClientLeaseBlock leaseBlock : leaseBlocks) {
+            if (!leaseBlock.matches(interfaceName, address)) {
+                continue;
+            }
+            List<IPAddress> routers = leaseBlock.getRouters();
+            for (IPAddress router : routers) {
+                if (!gateways.contains(router)) {
+                    gateways.add(router);
+                }
+            }
+        }
```

Comment:
Is it intentional to not break the outer loop on the fist match?

Reply:
Yes, I did this intentionally even though lease blocks appear to have the same information. I thought that if it is not, I will collect all routers/DNS servers.